src/api/eth_filter.rs

```diff
@@ -103,6 +107,20 @@ impl<T: Transport, R> BaseFilter<T, R> {
     CallResult::new(self.transport.execute("eth_getFilterChanges", vec![id]))
   }
 
+  pub fn stream<'a>(&'a self, poll_interval: Duration) -> Box<Stream<Item = R, Error = Error> + 'a> where
```

Comment:
Agree with consuming the filter! I will try prepare the struct, although it might not be trivial...